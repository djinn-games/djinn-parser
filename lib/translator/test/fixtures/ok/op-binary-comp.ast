{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "binary_comparisons",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "OperationExpression",
            "left": {
              "type": "Literal",
              "line": 3,
              "dataType": "int",
              "value": 2
            },
            "right": {
              "type": "Literal",
              "line": 3,
              "dataType": "int",
              "value": 2
            },
            "operator": "add",
            "line": 3
          },
          "right": {
            "type": "Literal",
            "line": 3,
            "dataType": "int",
            "value": 4
          },
          "operator": "gt",
          "line": 3
        },
        "line": 3
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "Literal",
            "line": 4,
            "dataType": "int",
            "value": 3
          },
          "right": {
            "type": "OperationExpression",
            "left": {
              "type": "Literal",
              "line": 4,
              "dataType": "int",
              "value": 2
            },
            "right": {
              "type": "Literal",
              "line": 4,
              "dataType": "int",
              "value": 1
            },
            "operator": "add",
            "line": 4
          },
          "operator": "eq",
          "line": 4
        },
        "line": 4
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "Literal",
            "line": 5,
            "dataType": "int",
            "value": 5
          },
          "right": {
            "type": "Literal",
            "line": 5,
            "dataType": "int",
            "value": 4
          },
          "operator": "lt",
          "line": 5
        },
        "line": 5
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "Literal",
            "line": 6,
            "dataType": "int",
            "value": 6
          },
          "right": {
            "type": "Literal",
            "line": 6,
            "dataType": "int",
            "value": 2
          },
          "operator": "gte",
          "line": 6
        },
        "line": 6
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "Literal",
            "line": 7,
            "dataType": "int",
            "value": 3
          },
          "right": {
            "type": "Literal",
            "line": 7,
            "dataType": "int",
            "value": 5
          },
          "operator": "neq",
          "line": 7
        },
        "line": 7
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "Literal",
            "line": 8,
            "dataType": "int",
            "value": 2
          },
          "right": {
            "type": "Literal",
            "line": 8,
            "dataType": "int",
            "value": 1
          },
          "operator": "lte",
          "line": 8
        },
        "line": 8
      }
    ],
    "line": 2
  },
  "line": 1
}