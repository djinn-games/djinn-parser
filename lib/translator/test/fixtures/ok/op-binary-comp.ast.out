{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "BinaryExpression",
        "operator": ">",
        "left": {
          "type": "BinaryExpression",
          "operator": "+",
          "left": {
            "type": "Literal",
            "value": 2,
            "raw": "2"
          },
          "right": {
            "type": "Literal",
            "value": 2,
            "raw": "2"
          }
        },
        "right": {
          "type": "Literal",
          "value": 4,
          "raw": "4"
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "BinaryExpression",
        "operator": "==",
        "left": {
          "type": "Literal",
          "value": 3,
          "raw": "3"
        },
        "right": {
          "type": "BinaryExpression",
          "operator": "+",
          "left": {
            "type": "Literal",
            "value": 2,
            "raw": "2"
          },
          "right": {
            "type": "Literal",
            "value": 1,
            "raw": "1"
          }
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "BinaryExpression",
        "operator": "<",
        "left": {
          "type": "Literal",
          "value": 5,
          "raw": "5"
        },
        "right": {
          "type": "Literal",
          "value": 4,
          "raw": "4"
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "BinaryExpression",
        "operator": ">=",
        "left": {
          "type": "Literal",
          "value": 6,
          "raw": "6"
        },
        "right": {
          "type": "Literal",
          "value": 2,
          "raw": "2"
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "BinaryExpression",
        "operator": "!=",
        "left": {
          "type": "Literal",
          "value": 3,
          "raw": "3"
        },
        "right": {
          "type": "Literal",
          "value": 5,
          "raw": "5"
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "BinaryExpression",
        "operator": "<=",
        "left": {
          "type": "Literal",
          "value": 2,
          "raw": "2"
        },
        "right": {
          "type": "Literal",
          "value": 1,
          "raw": "1"
        }
      }
    }
  ]
}
